### PR TITLE
Compose

### PR DESCRIPTION
**Pull request by mrnetops**
Docker Compose for a full fastly-exporter + Prometheus + Grafana + Fastly dashboard stack

```
FASTLY_API_TOKEN=${FASTLY_API_TOKEN} docker-compose up
```
Access the Grafana dashboard via http://localhost:3000

fastly-dashboard

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,3 +63,24 @@ services. Service IDs are available at the top of your [Fastly dashboard][db].
...
+[Docker Compose][compose] for a full fastly-exporter + [Prometheus][prom] + 
+[Grafana][grafana] + Fastly dashboard stack
+
+[compose]: https://github.com/docker/compose
+[grafana]: https://grafana.com
+
+```
+FASTLY_API_TOKEN=${FASTLY_API_TOKEN} docker-compose up
+```
+Access the [Grafana][grafana] dashboard via http://localhost:3000
```
Comment (peterbourgon) [inline README.md]:
```suggestion

Access the [Grafana][grafana] dashboard via http://localhost:3000.
```

File: README.md [hunk 2]
```diff
@@ -63,3 +63,24 @@ services. Service IDs are available at the top of your [Fastly dashboard][db].
...
+
+![Fastly Dashboard in Grafana](https://github.com/peterbourgon/fastly-exporter/compose/Fastly-Dashboard.png)
```
Comment (peterbourgon) [inline README.md]: Grotesquely pedantic nit: can you make this e.g. `dashboard.png` :)

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      - prometheus
+    ports:
+      - 3000:3000
+    volumes:
+      - ./compose/grafana/provisioning/:/etc/grafana/provisioning/
+      - grafana_data:/var/lib/grafana
+
+  fastly-exporter:
+    image: mrnetops/fastly-exporter
+    command: -token ${FASTLY_API_TOKEN}
```
Comment (peterbourgon) [inline docker-compose.yml]: You want to leverage the upcoming env var to avoid sticking the token in `ps` output?

**Comment by peterbourgon** [comment]
re: origin latency, isn't that [miss_duration_seconds](https://github.com/peterbourgon/fastly-exporter/blob/2e922a7505dd474cd1e03c6fb2b91f42727536ac/prometheus_metrics.go#L345)?

re: global POP traffic, I'm not sure exactly what you're after; can you say more? There is the synthetic "aggregated" datacenter, explained thusly:

> The keys of the datacenter JSON object are the names of each of our POPs around the world plus a special key, called aggregated, which adds all the values from all the POPs together.

The fastly-exporter collects that data, but doesn't do anything with it, because in Prometheus you can just sum over the datacenter dimension and get the same aggregation at query time.

File: docker-compose.yml (hunk 3 above)
Comment (mrnetops) [reply docker-compose.yml]: I'll adjust it to pure ENV when @ljagiello finishes their edits and we merge.  I suspect he'll be done before we finish out the review.

I know you're not a docker (or I assume docker-compose) fan but you should check this out, it's hella cool to see in action.

**Comment by mrnetops** [comment]
re: global POP traffic that just ties back into #16  I need the region or latitude / longitude to place stats on the map.  i.e. rt.fastly.com + api.fastly.com has the data.  I'm pretty sure I just need to get the data into fastly_rt_datacenter_info so I can pivot off of it.

**Comment by peterbourgon** [comment]
Oh I see it's going to be a map viz. Sure, do you know which Fastly API has that stuff? It should be easy to one-off it at program start. (Certainly not worth pinging the API every 5 minutes to see if POP locations have changed ;))

**Comment by mrnetops** [comment]
Aye! https://docs.fastly.com/api/tools#datacenter to populate fastly_rt_datacenter_info

I'm trying to replicate the standard fastly service dashboard in it's entirety for my initial canned dashboard.

**Comment by mrnetops** [comment]
re: origin latency.  ooooh, you may be onto something there!

Also, I'd love some thoughts on `HIT TIME` and `MISS TIME`, I'm having a hard time getting my dashboard to line up with the stock fastly dashboard for that.  `HIT TIME` seems to be an order of magnitude off, and `MISS TIME` is just weird.

**Comment by peterbourgon** [comment]
What is "the stock Fastly dashboard"?

**Comment by mrnetops** [comment]
https://manage.fastly.com/dashboard/services/<service_id>/datacenters/all

File: docker-compose.yml (hunk 3 above)
Comment (ljagiello) [reply docker-compose.yml]: @mrnetops it's done now.

Comment (mrnetops) [reply docker-compose.yml]: Docker hub updated (@peterbourgon we probably want to tag and release after this PR) and update added!

Nice addition and suggestion @ljagiello @peterbourgon

**Comment by mrnetops** [comment]
@peterbourgon let me know if you have any last edits, otherwise I'll merge the PR

**Review by peterbourgon — APPROVED** [review]

**Pull request merged**